primitives/src/v2/mod.rs

```diff
@@ -1689,6 +1689,35 @@ impl PvfCheckStatement {
 	}
 }
 
+/// Type discriminator for different PVF-related timeouts
+#[derive(Encode, Decode, TypeInfo, Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PvfTimeoutType {
+	/// For prechecking requests, the time period after which the preparation worker is considered
+	/// unresponsive and will be killed.
+	PrecheckPreparation,
+
+	/// For execution and heads-up requests, the time period after which the preparation worker is
+	/// considered unresponsive and will be killed. More lenient than the timeout for prechecking
+	/// to prevent honest validators from timing out on valid PVFs.
+	LenientPreparation,
```

Comment:
Looks like this variant is not used anywhere, AFAICT.

Reply:
Good catch, sounds like a real problem, that means I'm using a wrong timeout somewhere. Will investigate.

Reply:
Found it, I've mixed preparation and execution timeouts in several places which is really bad and makes me think that having separate `PvfPreparationTimeoutKind` and `PvfExecutionTimeoutKind` would be a good idea that could help us to avoid such errors in the future. The only drawback is that really long naming (as usual).

Reply:
If we agree to abbreviate "preparation" as "prep" and "execution" as "exec" everywhere, it would be less painful

Reply:
I am always in favor of using more types to ensure correctness in the code. ⭐️ 

I don't think those names are *that* long, pretty sure we have longer names than that, but I wouldn't mind shortening to "prep" and "exec" everywhere -- should be pretty evident what those mean.